packages/@aws-cdk/aws-events-targets/lib/util.ts

```diff
@@ -49,3 +50,45 @@ export function addLambdaPermission(rule: events.IRule, handler: lambda.IFunctio
     });
   }
 }
+
+/**
+ * Allow a rule to send events with failed invocation to an Amazon SQS queue.
+ */
+export function addToDeadLetterQueueResourcePolicy(rule: events.IRule, queue: sqs.IQueue) {
+  if (!sameEnvDimension(rule.env.region, queue.env.region)) {
+    throw new Error(`Cannot assign Dead Letter Queue in region ${queue.env.region} to the rule ${Names.nodeUniqueId(rule.node)} in region ${rule.env.region}. Both the queue and the rule must be in the same region.`);
+  }
+
+  // Skip Resource Policy creation if the Queue is not in the same account.
+  // There is no way to add a target onto an imported rule, so we can assume we will run the following code only
+  // in the account where the rule is created.
```

Comment:
There is also a common use case where a stack creates a rule on a bus in another account. Will this work when the rule is in a different account than the target and the queue?

https://aws.amazon.com/blogs/compute/simplifying-cross-account-access-with-amazon-eventbridge-resource-policies/

Reply:
Actually, looks like it will work based on the blog (last role).

But it does mean that the rule and the queue can be in separate accounts. The comment is just not quite accurate.